src/Kiota.Builder/Extensions/OpenApiSchemaExtensions.cs

```diff
@@ -121,15 +123,26 @@ public static bool IsODataPrimitiveType(this OpenApiSchema schema)
     }
     public static bool IsEnum(this OpenApiSchema schema)
     {
-        return (schema?.Enum?.Any() ?? false) && (string.IsNullOrEmpty(schema.Type) || "string".Equals(schema.Type, StringComparison.OrdinalIgnoreCase)); // number and boolean enums are not supported
+        return (schema?.Enum?.Any() ?? false) &&
+                (string.IsNullOrEmpty(schema.Type) || "string".Equals(schema.Type, StringComparison.OrdinalIgnoreCase)) ||
+                (schema?.AnyOf.Where(static x => x.IsSemanticallyMeaningful(true)).Any(static x => x.IsEnum()) ?? false) ||
+                (schema?.OneOf.Where(static x => x.IsSemanticallyMeaningful(true)).Any(static x => x.IsEnum()) ?? false)
+        ; // number and boolean enums are not supported
     }
     public static bool IsComposedEnum(this OpenApiSchema schema)
     {
         return (schema.IsInclusiveUnion() && schema.AnyOf.Any(static x => x.IsEnum())) || (schema.IsExclusiveUnion() && schema.OneOf.Any(static x => x.IsEnum()));
     }
-    private static bool IsSemanticallyMeaningful(this OpenApiSchema schema)
+    private static bool IsSemanticallyMeaningful(this OpenApiSchema schema) => IsSemanticallyMeaningful(schema, false);
```

Comment:
Just to confirm here, this could be used as a optional parameter but the compilation would break due to the function being called as a method group in some instances.

Reply:
yep, I was too lazy to add lambdas at all the places it's used in LINQ statements just by passing the reference to the method. But I can align on an optional parameter if you think the code would be clearer this way

Reply:
Thanks for making the change!